test/apps/Makefile.am

```diff
@@ -6,8 +6,10 @@
 # See file LICENSE for terms.
 #
 
+
 if HAVE_CXX11
-SUBDIRS = sockaddr
+SUBDIRS = iodemo
```

Comment:
can we fix the code to work w/o C11? IMHO would make it easier for users to use it as base for their app

Reply:
Correct and portable way for all compilers and standards - no. The problem is in `noexcept`/`throw()` specification. `noexcept` is added in c++11, `throw()` is deprecated in c++11 and removed in c++20.
If it's really needed we can do a trick like `_GLIBCXX_NOEXCEPT` macro but IMHO c++11 is old enough to be used in modern example

Reply:
can you use _GLIBCXX_USE_NOEXCEPT ?

Reply:
gcc (GCC) 4.4.7 20120313 does not have this macro.

Reply:
added configure test of `_GLIBCXX_NOTHROW` presence for the app compilation